front_end/src/app/(main)/questions/[id]/components/forecast_maker/prediction_status_message.tsx

```diff
@@ -0,0 +1,67 @@
+import { useTranslations } from "next-intl";
+import { FC } from "react";
+
+import { PostStatus, PostWithForecasts, QuestionStatus } from "@/types/post";
+
+type Props = {
+  post: PostWithForecasts;
+};
+
+const PredictionStatusMessage: FC<Props> = ({ post }) => {
+  const t = useTranslations();
+
+  switch (post.status) {
+    case PostStatus.UPCOMING: {
+      return <>{t("predictionUpcomingMessage")}</>;
+    }
+    case PostStatus.APPROVED: {
+      if (Date.parse(post.open_time) > Date.now()) {
+        return <>{t("predictionUpcomingMessage")}</>;
+      }
+      return null;
+    }
+    case PostStatus.REJECTED:
+    case PostStatus.PENDING:
+    case PostStatus.DRAFT: {
+      return <>{t("predictionUnapprovedMessage")}</>;
+    }
+    case PostStatus.CLOSED: {
+      if (!post.resolved) {
+        if (post.conditional) {
+          const questions = [
+            post.conditional.condition,
+            post.conditional.condition_child,
+          ];
+          const targetStatuses = [
+            QuestionStatus.CLOSED,
+            QuestionStatus.RESOLVED,
+          ];
+
+          const closedQuestion = questions.find((obj) =>
+            targetStatuses.includes(obj.status!)
+          );
+          const activeQuestion = questions.find(
+            (obj) => !targetStatuses.includes(obj.status!)
+          );
+
+          if (closedQuestion && activeQuestion) {
+            return (
+              <>
+                {t("predictionConditionalPartiallyClosedMessage", {
+                  closedQuestion: closedQuestion.title,
+                  activeQuestion: activeQuestion.title,
+                })}
```

Comment:
I don't think there should ever be a state where there is one close and one open branch?
What situation are you imagining?

Reply:
If we're already in the PostStatus.CLOSED situation, then how would there be an active branch question?

Reply:
This just seems like unreachable code, so not blocking, just unnecessary

Reply:
@lsabor We have an example of post with closed child quesiton -- https://www.metaculus.com/questions/22051/starship-booster-tower-catch-attempt-in-2024/

https://github.com/Metaculus/metaculus/issues/1008